### PR TITLE
INT-355_test_iq_server_dockerfile

### PR DESCRIPTION
**Pull request by fernau**
JIRA: https://issues.sonatype.org/browse/INT-355

I would prefer serverspec.org for testing docker images. It is essentially the same what we do for chef, but I found kitchen not to work very well with docker.

- serverspec does not need SSH in the docker container
- serverspec does not hijack the 'CMD' command to run its tests
- serverspec runs the tests inside the container

### CODE REVIEW

File: spec/Dockerfile_spec.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# spec/Dockerfile_spec.rb
```
Comment (whyjustin) [inline spec/Dockerfile_spec.rb]: License header

File: spec/Dockerfile_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    Docker.options[:read_timeout] = 900
+    @image = Docker::Image.build_from_dir('.')
+
+    set :os, family: :redhat
+    set :backend, :docker
+    set :docker_image, @image.id
+  end
+
+  it "solo.json file is valid" do
+    solo = JSON.parse(File.read('/var/chef/solo.json'))
```
Comment (whyjustin) [inline spec/Dockerfile_spec.rb]: Single or double quotes consistently

File: spec/Dockerfile_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    expect(solo['run_list']).to eq([ 'recipe[nexus-iq-server::docker]' ])
+  end
+
+  it "chef package is installed" do
+    expect(package("chef")).to be_installed
+  end
+
+  it "nexus java process is running" do
+    expect(process('java')).to be_running
+    expect(process('java')[:user]).to eq('nexus')
```
Comment (whyjustin) [inline spec/Dockerfile_spec.rb]: Can we verify more with the process args? Or just the user?

**Review by whyjustin — CHANGES_REQUESTED** [review]

File: spec/Dockerfile_spec.rb (hunk 3 above)
Comment (fernau) [reply spec/Dockerfile_spec.rb]: we can verify most things with process. here is the link: http://serverspec.org/resource_types.html#process

File: spec/Dockerfile_spec.rb (hunk 1 above)
Comment (fernau) [reply spec/Dockerfile_spec.rb]: added license header in 0007ead

File: spec/Dockerfile_spec.rb (hunk 2 above)
Comment (fernau) [reply spec/Dockerfile_spec.rb]: fixed inconsistent use of quotes in 0007ead

File: spec/Dockerfile_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    Docker.options[:read_timeout] = 900
+    @image = Docker::Image.build_from_dir('.')
+
+    set :os, family: :redhat
+    set :backend, :docker
+    set :docker_image, @image.id
+  end
+
+  it 'should generate a valid solo.json file' do
+    solo = JSON.parse(File.read('/var/chef/solo.json'))
```
Comment (bigspotteddog) [inline spec/Dockerfile_spec.rb]: This file will not exist when after the cleanup happens. I added a change in #3 to address this.

**Review by bigspotteddog — APPROVED** [review]

**Review by whyjustin — APPROVED** [review]

**Pull request merged**